apis/s3/src/main/java/org/jclouds/s3/S3Client.java

```diff
@@ -640,6 +651,30 @@ void enableBucketLogging(@Bucket @EndpointParam(parser = AssignCorrectHostnameFo
    void disableBucketLogging(@Bucket @EndpointParam(parser = AssignCorrectHostnameForBucket.class) @BinderParam(
                BindNoBucketLoggingToXmlPayload.class) @ParamValidators(BucketNameValidator.class) String bucketName);
 
+   @Named("GetBucketConfiguration")
+   @GET
+   @Path("/")
+   @ResponseParser(BucketConfigurationHandler.class)
+   String getBucketConfiguration(@Bucket @EndpointParam(parser = AssignCorrectHostnameForBucket.class) @BinderParam(
+       BindAsHostPrefixIfConfigured.class) @ParamValidators(BucketNameValidator.class) String bucketName,
+                                    BucketConfigOptions... options);
+
+   @Named("DeleteBucketConfiguration")
+   @DELETE
+   @Path("/")
+   @ResponseParser(BucketConfigurationHandler.class)
+   String deleteBucketConfiguration(@Bucket @EndpointParam(parser = AssignCorrectHostnameForBucket.class) @BinderParam(
+       BindAsHostPrefixIfConfigured.class) @ParamValidators(BucketNameValidator.class) String bucketName, BucketConfigOptions... options);
+
+   @Named("PutBucketConfiguration")
+   @PUT
+   @Path("/")
+   @ResponseParser(BucketConfigurationHandler.class)
+   String putBucketConfiguration(@Bucket @EndpointParam(parser = AssignCorrectHostnameForBucket.class) @BinderParam(
```

Comment:
Can you pull the bucket configuration into a separate PR with unit tests?

Reply:
Hi @gaul 
Will be raising new PR for this change with unit tests, and other noise changes highlighted in above comments.